src/rust/engine/src/nodes.rs

```diff
@@ -423,34 +437,80 @@ impl ExecuteProcess {
       timeout: Duration::from_millis((timeout_in_seconds * 1000.0) as u64),
       description: description,
       jdk_home: jdk_home,
-    }))
+    })
+  }
+
+  fn lift(value: &Value) -> Result<MultiPlatformExecuteProcess, String> {
+    let mut request_by_constraint: BTreeMap<(Platform, Platform), ExecuteProcessRequest> =
+      BTreeMap::new();
+    let constraint_parts = externs::project_multi_strs(&value, "platform_constraints");
+    let mut requests: Vec<ExecuteProcessRequest> = Vec::new();
+    for execute_process in externs::project_multi(&value, "execute_process_requests").iter() {
+      let underlying_req = MultiPlatformExecuteProcess::lift_execute_process(execute_process)?;
+      requests.push(underlying_req);
+    }
+    if constraint_parts.len() % 2 != 0 {
+      return Err("Error parsing platform_constraints: odd number of parts".to_owned());
+    }
+    if constraint_parts.len() / 2 != requests.len() {
+      return Err(format!(
+        "Size of constraint keys and requests does not match: {} vs. {}",
+        constraint_parts.len() / 2,
+        requests.len()
+      ));
+    }
+    for (constraint_key_pair, request) in constraint_parts.chunks_exact(2).zip(requests.iter()) {
```

Comment:
Maps are actually `collect`able-to from two-tuples, so you should be able to phrase this more as:
```rust
    let request_by_constraint = constraint_parts.chunks_exact(2).map(|constraint_key_pair| {
      // Turn constraint_key_pair into (platform, platform)
    })
    .zip(requests.into_iter())
    .collect();
```

Reply:
cool!